frame/staking/src/lib.rs

```diff
@@ -3382,17 +3395,22 @@ impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
 				}
 			}
 
+			// discard solution if it is too big.
+			if !Self::check_unsigned_solution_weight(winners, compact, size) {
+				return Err(InvalidTransaction::ExhaustsResources.into())
```

Comment:
I think returning this error is missleading:
the doc of ExhaustResources is:
```
	/// The transaction would exhaust the resources of current block.
	///
	/// The transaction might be valid, but there are not enough resources
	/// left in the current block.
```
I'm not sure which error to use, but for me submitting a solution which overflow limit is just and invalid call.

Reply:
outdated.